specs/commandline/blob.md

```diff
@@ -0,0 +1,508 @@
+
+```text
+Import blob trust policy configuration from a JSON file
+
+Usage:
+  notation blob policy import [flags] <file_path>
+
+Flags:    
+      --force     override the existing trust policy configuration, never prompt
+  -h, --help      help for import
+```
+
+### notation blob policy show
+
+```text
+Show blob trust policy configuration
+
+Usage:
+  notation blob policy show [flags]
+
+Flags:
+  -h, --help      help for show
+```
+
+### notation blob verify
+
+```text
+Verify a signature associated with a blob
+
+Usage:
+  notation blob verify [flags] --signature <signature_path> <blob_path>
+
+Flags:
+      --signature string      location of the blob signature file
+      --media-type string     optional media type of the blob to verify
+      --policy-name string    optional policy name to verify against. If not provided, notation verifies against the global policy if it exists.
+  -m, --user-metadata stringArray   user defined {key}={value} pairs that must be present in the signature for successful verification if provided
+      --plugin-config stringArray   {key}={value} pairs that are passed as it is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values
+  -o, --output string         output format, options: 'json', 'text' (default "text")
+  -d, --debug                 debug mode
+  -v, --verbose               verbose mode
+  -h, --help                  help for inspect
+```
+
+## Usage
+
+## Produce blob signatures
+
+### Sign a blob by adding a new key
+
+```shell
+# Prerequisites:
+# - A signing plugin is installed. See plugin documentation (https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details.
+# - Configure the signing plugin as instructed by plugin vendor.
+
+# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
+notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
+
+# sign a blob
+notation blob sign /tmp/my-blob.bin
+```
+
+An example for a successful signing:
+
+```console
+$ notation blob sign /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to /tmp/my-blob.bin.sig.jws
+```
+
+### Sign a blob by generating the signature in a particular directory
+```console
+$ notation blob sign --signature-directory /tmp/xyz/sigs /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to /tmp/xyz/sigs/my-blob.bin.sig.jws
+```
+
+### Sign a blob using a relative path
+```console
+$ notation blob sign ./relative/path/my-blob.bin
+Successfully signed ./relative/path/my-blob.bin
+Signature file written to ./relative/path/my-blob.bin.sig.jws
+```
+
+### Sign a blob and skip user confirmations when overwriting existing signature
+```console
+$ notation blob sign --force /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file overwritten to /tmp/my-blob.bin.sig.jws
+```
```

Comment:
Here comes the inconsistency.

The definition above
>        --signature-directory string optional path where the blob signature needs to be placed (default: currently working directory) 

states that the signature should be default written to the current working directory.

However, the above two examples are not (so as other examples).

Reply:
Shouldn't it be

------

### Sign a blob using a relative path
```console
$ notation blob sign ./relative/path/my-blob.bin
Successfully signed ./relative/path/my-blob.bin
Signature file written to my-blob.bin.sig.jws
```

### Sign a blob and skip user confirmations when overwriting existing signature
```console
$ notation blob sign --force /tmp/my-blob.bin
Successfully signed /tmp/my-blob.bin
Signature file overwritten to my-blob.bin.sig.jws
```

------

or print the absolute path given the current working directory `/test`

### Sign a blob using a relative path
```console
$ notation blob sign ./relative/path/my-blob.bin
Successfully signed ./relative/path/my-blob.bin
Signature file written to /test/my-blob.bin.sig.jws
```

### Sign a blob and skip user confirmations when overwriting existing signature
```console
$ notation blob sign --force /tmp/my-blob.bin
Successfully signed /tmp/my-blob.bin
Signature file overwritten to /test/my-blob.bin.sig.jws
```

Reply:
I like the absolute path option. Makes it less ambiguous. Will update with this.